client/src/components/ShowsList.js

```diff
@@ -4,6 +4,18 @@ import ShowTile from "./ShowTile.js"
 
 const ShowsList = (props) => {
   const [shows, setShows] = useState([])
+  const [searchData, setSearchData] = useState("")
+
+  let searchedShows = []
+  if(shows.length > 0) {
+    shows.forEach(show => {
+      if(show.name.toLowerCase().includes(searchData.toLowerCase()) && !searchedShows.includes(show)) {
+        searchedShows.push(show)
+      } else if (!show.name.toLowerCase().includes(searchData.toLowerCase()) && searchedShows.includes(show)) {
+        searchedShows = searchedShows.filter(searchedShow => searchedShow !== show)
+      }
+    })
+  }
```

Comment:
Two things:

1. This implementation is totally fine, but would likely be slightly more powerful if you executed the search on the backend. Up to you. If we had a database of over 1000 shows, we would want to execute the search in the database rather than hear in react, as it would be more efficient and we wouldnt need to hold all of our shows in state. 

2. I would check out potentially refactoring with the `.find` method. 

3. I also am not sure why we are running `searchedShows.includes(show)`. Maybe Im missing something? Are there duplicates of a show? 

4. In any case, we can dry this up. If we get to the second conditional, then for every show we will have checked if it contains your search query twice, and we will have checked `searchedShows` to see if the given show is included twice. 

Instead, try to only compute these things once. 

```
  shows.forEach(show => {
       const showIsPresent = show.name.toLowerCase().includes(searchData.toLowerCase()
       const showAlreadyInList = searchedShows.includes(show)) 
       
       ....conditional logic
   
    })
```

Reply:
The purpose of searchedShows.includes(show) is to determine whether a show has already been added to the list.  If we didn't do that, if you type "g", then "a", then "m", Game of Thrones would be pushed onto the array of searchedShows three times in a row.